src/plugins/visualizations/server/saved_objects/visualization_migrations.ts

```diff
@@ -981,6 +982,96 @@ const removeDefaultIndexPatternAndTimeFieldFromTSVBModel: SavedObjectMigrationFn
   return doc;
 };
 
+const addEmptyValueRuleForSavedObjectsWithLessAndGreaterThenZeroRules: SavedObjectMigrationFn<
+  any,
+  any
+> = (doc) => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+
+    if (visState && visState.type === 'metrics') {
+      const params: any = get(visState, 'params') || {};
+      const {
+        bar_color_rules: barColorRules = [],
+        background_color_rules: backgroundColorRules = [],
+        gauge_color_rules: gaugeColorRules = [],
+      } = params;
+
+      const compareWithEqualMethods = ['lte', 'gte'];
+      const emptyRule = {
+        operator: 'empty',
+        value: null,
+      };
+
+      const getRulesWithComparingToZero = (rules: any[] = []) =>
+        last(
+          rules.filter(
+            (rule) => compareWithEqualMethods.includes(rule.operator) && rule.value === 0
+          )
+        );
+
+      const convertRuleToEmpty = ({ id, ...rest }: any = {}) => ({
+        ...rest,
+        ...emptyRule,
```

Comment:
nit: what do you think about moving `emptyRule -> convertRuleToEmpty` and `compareWithEqualMethods -> getRulesWithComparingToZero`.  
It seems to me that if we already have functions responsible for something, all associated code should be inside